File: kotlin-springboot-jpa/app/src/main/java/es/msanchez/frameworks/spring/boot/dao/PersonDao.java
package es.msanchez.frameworks.spring.boot.dao;

import es.msanchez.frameworks.spring.boot.entity.Person;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface PersonDao extends RawDao<Person> {

    @Query("SELECT p FROM Person p WHERE p.name = :name")
    Optional<Person> findOneByName(@Param("name") final String name);

    List<Person> findAllByAge(final Integer age);

}

File: kotlin-springboot-jpa/app/src/main/java/es/msanchez/frameworks/spring/boot/dao/PersonDao.kt
package es.msanchez.frameworks.spring.boot.dao

import es.msanchez.frameworks.spring.boot.entity.Person
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface PersonDao : RawDao<Person> {

    @Query("SELECT p FROM Person p WHERE p.name = :name")
    fun findOneByName(@Param("name") name: String): Optional<Person>

    fun findAllByAge(age: Int): List<Person>

}